x-pack/plugin/stack/src/javaRestTest/java/org/elasticsearch/xpack/stack/EcsDynamicTemplatesIT.java

```diff
@@ -0,0 +1,400 @@
+import static org.hamcrest.Matchers.instanceOf;
+
+@SuppressWarnings("unchecked")
+public class EcsDynamicTemplatesIT extends ESRestTestCase {
+
+    @ClassRule
+    public static ElasticsearchCluster cluster = ElasticsearchCluster.local().module("mapper-extras").module("wildcard").build();
+
+    // The dynamic templates we test against
+    public static final String ECS_DYNAMIC_TEMPLATES_FILE = "/ecs-dynamic-mappings.json";
+
+    // The current ECS state (branch main) as a containing all fields in flattened form
+    private static final String ECS_FLAT_FILE_URL = "https://raw.githubusercontent.com/elastic/ecs/main/generated/ecs/ecs_flat.yml";
+
+    private static final Set<String> OMIT_FIELD_TYPES = Set.of("object", "flattened", "nested");
+
+    private static final Set<String> OMIT_FIELDS = Set.of("data_stream.dataset", "data_stream.namespace", "data_stream.type");
+
+    private static Map<String, Object> ecsDynamicTemplates;
+    private static Map<String, Map<String, Object>> ecsFlatFieldDefinitions;
+    private static Map<String, String> ecsFlatMultiFieldDefinitions;
+
+    @BeforeClass
+    public static void setupSuiteScopeCluster() throws Exception {
+        prepareEcsDynamicTemplates();
+        prepareEcsDefinitions();
+    }
+
+    private static void prepareEcsDynamicTemplates() throws IOException {
+        String rawEcsComponentTemplate = TemplateUtils.loadTemplate(
+            ECS_DYNAMIC_TEMPLATES_FILE,
+            Integer.toString(1),
+            StackTemplateRegistry.TEMPLATE_VERSION_VARIABLE,
+            Collections.emptyMap()
+        );
+        Map<String, Object> ecsDynamicTemplatesRaw;
+        try (
+            XContentParser parser = XContentFactory.xContent(XContentType.JSON)
+                .createParser(XContentParserConfiguration.EMPTY, rawEcsComponentTemplate)
+        ) {
+            ecsDynamicTemplatesRaw = parser.map();
+        }
+
+        String errorMessage = String.format(
+            Locale.ENGLISH,
+            "ECS mappings component template '%s' structure has changed, this test needs to be adjusted",
+            ECS_DYNAMIC_TEMPLATES_FILE
+        );
+        assertFalse(errorMessage, rawEcsComponentTemplate.isEmpty());
+        Object mappings = ecsDynamicTemplatesRaw.get("template");
+        assertNotNull(errorMessage, mappings);
+        assertThat(errorMessage, mappings, instanceOf(Map.class));
+        Object dynamicTemplates = ((Map<?, ?>) mappings).get("mappings");
+        assertNotNull(errorMessage, dynamicTemplates);
+        assertThat(errorMessage, dynamicTemplates, instanceOf(Map.class));
+        assertEquals(errorMessage, 1, ((Map<?, ?>) dynamicTemplates).size());
+        assertTrue(errorMessage, ((Map<?, ?>) dynamicTemplates).containsKey("dynamic_templates"));
+        ecsDynamicTemplates = (Map<String, Object>) dynamicTemplates;
+    }
+
+    @SuppressForbidden(reason = "Opening socket connection to read ECS definitions from ECS GitHub repo")
```

Comment:
Note for reviewer: what would be the right way to download a file from GitHub? For now, I just used `@SuppressForbidden` to be able to test

Reply:
The better way is probably to declare a custom ivy repo in the build script and have Gradle download it so that' it is cached rather than downloading on every test execution.

Reply:
Thanks for the proposal! 
Since it is a single file that we download, is adding a dependency on the full repo indeed preferable? What I meant to ask is whether relying on TLS for this single file download good enough.
I am not familiar enough with Ivy- does it require specifying a version of the target repo? If so- this won't address what we need, which is to see ECS changes as soon as they are made. 

If you still think we should replace this, please provide an example that I can use as reference.

Reply:
You can more or less hard-code the URL in the repo definition. Here's an example we use for [downloading github releases](https://github.com/elastic/elasticsearch/blob/cc2a711b2de2800a34bc34bcc89a4b101485ab37/distribution/docker/build.gradle#L31), but you can adapt it to just download a raw file easily. The placeholders here are for maven artifact ids that look like `[org]:[module]:[revision]`.

Reply:
I'm not sure if that would translate well to this case, though, as we're downloading a file from main, not from a particular version. Therefore, caching would be difficult, too, unless GitHub supports conditional GETs. But I guess what we could do is to simply download the file in via gradle instead of within the unit test. Something similar to `file("${project.buildDir}/ecs-dynamic-mappings.json") << "https://raw.githubusercontent.com/elastic/ecs/main/generated/ecs/ecs_flat.yml".toUrl().text`.

However, that doesn't have many benefits over just downloading the file directly from the unit test itself, except from not having to apply the `@SuppressForbidden` annotation.

Reply:
We can follow up here. You can set a cache invalidation strategy in the build so it'll redownload but we can attack this at a later time.

Reply:
So for the time being, are we good with the current approach?

Reply:
@mark-vieira can I mark this one as resolved? Are you OK with this simple HTTPS file download from GitHub?

Reply:
Yes, that's fine.